tools/evmc/main.cpp

```diff
@@ -10,25 +10,27 @@
 
 namespace
 {
-/// Returns the input str if already valid hex string. Otherwise, interprets the str as a file
-/// name and loads the file content.
+/// If the argument starts with @ returns the content of the file at the path following @.
+/// Otherwise, returns the argument.
 /// @todo The file content is expected to be a hex string but not validated.
 std::string load_hex(const std::string& str)
 {
-    if (evmc::validate_hex(str))
-        return str;
+    if (str[0] == '@')  // The argument is file path.
+    {
+        std::ifstream file{str.c_str() + 1};
```

Comment:
Why use `c_str() + 1` here and `substr(1)` below?

Reply:
I actually changed this to add more info to the error.